src/modules/session-recorder/SessionRecorder.ts

```diff
@@ -281,4 +287,41 @@ export class SessionRecorder {
 
     return { ...networkEventData, errors };
   }
+
+  #captureInitialLocalStorageDump(): void {
+    const localStorageKeys = Object.keys(localStorage);
+    localStorageKeys.forEach((key) => {
+      const value = localStorage?.getItem(key);
+      const storageEvent: StorageEventData = {
+        timestamp: Date.now(),
+        key,
+        value,
+        eventType: "initialStorageValue",
+        };
+        console.log("storageEvent", storageEvent);
+        this.#addEvent(RQSessionEventType.STORAGE, storageEvent);
+    });
+  }
+
+  #startCapturingLocalStorage(): void {
+    window.addEventListener('storage', this.#captureStorageEvent);
+  }
+
+  #stopCapturingLocalStorage(): void {
+    window.removeEventListener('storage', this.#captureStorageEvent);
+  }
+
+  #captureStorageEvent = (event: StorageEvent): void => {
+    if (event.storageArea === localStorage) {
+      const storageEvent: StorageEventData = {
+        timestamp: Date.now(),
+        key: event.key,
+        eventType: "keyUpdate",
+        oldValue: event.oldValue,
+        newValue: event.newValue,
+      };
+      this.#addEvent(RQSessionEventType.STORAGE, storageEvent);
+    }
+  };
```

Comment:
It should take a parameter which storage area to capture based on `SessionRecorderOptions`